src/bacnet/basic/object/nc.c

```diff
@@ -249,33 +249,37 @@ int Notification_Class_Read_Property(BACNET_READ_PROPERTY_DATA *rpdata)
             break;
 
         case PROP_RECIPIENT_LIST:
-            /* get the size of all entry of Recipient_List */
-            for (idx = 0; idx < NC_MAX_RECIPIENTS; idx++) {
+            {
+              /* get the size of all entry of Recipient_List */
+              int apdu_test_len = apdu_len;
```

Comment:
Why can't you just use apdu_len? Just curious as to why this is needed

Reply:
On behalf of @metelik :

"  because apdu_len is returned by the ReadProperty handler, when it is increased by both: test of eventual length and actual read prop' the returned apdu_len' is actual apdu_len + test of 10 entries len and lands in the actual tag APDU len of the BACnet packet - detected as the malformed."